exporters/otlp/otlpmetric/otlpmetricgrpc/client.go

```diff
@@ -77,6 +78,8 @@ func newClient(ctx context.Context, options ...Option) (otlpmetric.Client, error
 	}
 
 	if c.conn == nil {
+		// Add a User-Agent header when no ClientConn was provided
+		cfg.DialOptions = append(cfg.DialOptions, grpc.WithUserAgent(internal.GetUserAgentHeader()))
```

Comment:
If the user of the library is passing their own user agent (Here, in `cf.DialOptions`, for HTTP in `cf.Headers`), the behaviors between HTTP and gRPC differ.

In gRPC, we append the user agent to the passed options. So our value will always take precedence.
In HTTP, we set our header before setting the header options. So the user's value will take precedence.

I think the behavior should be the same for both protocols, and documented/tested.
My preference would go for respecting the user's options. If they wish to override the user-agent, they should be able to do so.

Reply:
Totally agree with you, move it to init, will allow user override it.

Reply:
Could you add tests as well for this behavior?

Reply:
ok, I add some tests for this.